cpp/src/ray/config_internal.cc

```diff
@@ -78,9 +94,23 @@ void ConfigInternal::Init(RayConfig &config, int *argc, char ***argv) {
     }
     gflags::ShutDownCommandLineFlags();
   }
-  if (worker_type == WorkerType::DRIVER) {
-    RAY_CHECK(run_mode == RunMode::SINGLE_PROCESS || !dynamic_library_path.empty())
-        << "Please add a local dynamic library by '--ray-dynamic-library-path'";
+  if (worker_type == WorkerType::DRIVER && run_mode == RunMode::CLUSTER) {
+    if (code_search_path.size() == 0) {
+      auto program_path = boost::dll::program_location().parent_path();
+      RAY_LOG(WARNING) << "No code search path found yet. "
```

Comment:
Please make it not a warning because it's totally valid use.

Reply:
ok